File: BOJ/11659.py
import sys
input = sys.stdin.readline

n, m = map(int, input().split())
lst = list(map(int, input().split()))
prefix_sum = [0]

temp = 0

for i in lst:
    temp += i
    prefix_sum.append(temp)
 
for i in range(m):
    a, b = map(int, input().split())
    print(prefix_sum[b] - prefix_sum[a-1])

File: javascript/boj/prefix-sum/11659/11659.js
const path = process.platform === 'linux' ? '/dev/stdin' : 'input.txt'
const input = require('fs').readFileSync(path).toString().trim().split('\n');

const [N, M] = input[0].split(' ').map(Number);
const nums = input[1].split(' ').map(Number);

const prefixSum = new Array(N+1).fill(0);
const res = [];

nums.forEach((n, i) => {
  prefixSum[i+1] = prefixSum[i] + n;
});


input.slice(2).forEach(tc => {
  const [a, b] = tc.split(' ').map(Number);
  res.push(prefixSum[b]-prefixSum[a-1]);
});

console.log(res.join('\n'));